diagnostic_updater/include/diagnostic_updater/DiagnosticStatusWrapper.hpp

```diff
@@ -140,10 +145,13 @@ class DiagnosticStatusWrapper : public diagnostic_msgs::DiagnosticStatus
   void mergeSummaryf(unsigned char lvl, const char * format, ...)
   {
     va_list va;
-    char buff[1000];     // @todo This could be done more elegantly.
+    const int kBufferSize = 1000;
```

Comment:
that was the only way I've seen so far which seemed to make the ros2 linters happy. Hope that's okay for you as well.